source/4.0/learn/introduction/active-record.html.md

```diff
@@ -1,82 +1,545 @@
+  end
+end
+```
+
+### Mapping Custom Models
+
+```ruby
+class CustomUser < MySuperModelLibary
+end
+
+users_relation.map_to(CustomUser).first
+
+#> #<CustomUser id="1", username="Joe">
+```
+
+ROM does not care what your final output object is as long as it accepts a hash
+of all the attributes and their values. Coupled with other mappers, the output
+from a query can be incredibly flexible.
+
+### SQL Functions
+
+<h4 class="text-center">Active Record</h4>
+```ruby
+User
+  .select("*, concat(first_name, ' ', last_name) as 'full_name')")
+  .first
+
+#> #<User id=1 full_name="Jane Doe">
+```
+
+
+<h4 class="text-center">ROM</h4>
+```ruby
+users_relation.select_append {
+  str::first_name.concat(' ', last_name).as(:full_name)
+}.first
+
+#> #<ROM::Struct::User id=1, full_name="Jane Doe">
+```
+
+### Legacy Schemas
+
+<h4 class="text-center">Active Record</h4>
+```ruby
+class User < ApplicationRecord
+  self.table_name = 'SomehorriblyNamedUseRtable'
+  self.primary_key = 'UserIdentifier'
+
+  alias_attribute :id, :UserIdentifier
+  alias_attribute :name, :UserName
+end
+
+User.find_by(name: 'Jane')
+
+#> #<User UserIdentifier: "2", UserName: "Jane">
+
+User.where('name IS ?', 'Jane').first
+
+# 🔥🔥 KA-BOOM! 🔥🔥
+# ActiveRecord::StatementInvalid: no such column
+```
+
+
+<h4 class="text-center">ROM</h4>
+```ruby
+class Users < ROM::Relation[:sql]
+  schema(:SomehorriblyNamedUseRtable, as: :users) do
+    attribute :UserIdentifier, Serial.meta(alias: :id)
+    attribute :UserName, String.meta(alias: :name)
+  end
+end
+
+users_relation.where(name: 'Jane').first
+
+#> #<ROM::Struct::User id=1 name="Jane">
+```
+
+ROM makes working with legacy schemas a breeze. All that's needed it to define
+attributes on the relations schema along with
+return types and aliases. Afterwards just reuse the aliased names throughout
+your ROM queries - *quick* and *easy*.
+
+Working with ActiveRecord in this regard is a bit more difficult. While you
+can alias attributes, there is no real supported method for changing attribute
+names. Worse yet, ActiveRecord breaks the rule of Least Surprise because while
+some parts of the ActiveRecord API takes `alias_attribute` into account,
+[arel](https://github.com/rails/arel)
+does not, causing performance tuning SQL queries to fall back on the
+ugly database attribute names you were trying to avoid.
+
+### Custom Mappers
 
-Mixes domain-specific data validation with persistence layer. An active record
-object validates itself using its own validation rules.
+```ruby
+class EncryptionMapper < ROM::Mapper
+  register_as :encryption
 
-#### ROM
+  def call(relation)
+    relation.map {|tuple|
+      # do whatever you want
+    }
+  end
+end
 
-There's no validation concept built-in. Validations are handled externally by
-separate libraries and validated data can be passed down to the command layer to
-be persisted.
+users.map_with(:encryption)
+```
 
-## Data Coercion
+### Transform Data Before Persisting
```

Comment:
This section could use a bit of introduction. Changesets with data mapping is a first class feature, and personally I find it to be REALLY DAMN USEFUL, and it's something AR does not have at all :)

Reply:
I've written up a little diddy for the beginning here but if you have some examples to flesh out the section a bit more that  would be super helpful.